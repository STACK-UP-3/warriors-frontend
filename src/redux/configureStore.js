import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import promise from 'redux-promise-middleware';
import { composeWithDevTools } from 'redux-devtools-extension';
import reducersCombination from './reducers/index';



export default ()=>{
    const store = createStore(
        reducersCombination, {}, 
        composeWithDevTools(applyMiddleware(thunk,promise))
    );
    
    return store;
}
