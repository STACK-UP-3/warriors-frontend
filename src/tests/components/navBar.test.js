import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import Navbar from '../../components/navBar';
import configureStore from '../../redux/configureStore';

describe('*************** Testing the Navbar  component ***************',()=>{
    const store = configureStore();

    it('Should render the Navbar page correctly', ()=>{
        const wrapper = mount( 
            <Provider store={ store }> 
                <Navbar  /> 
            </Provider>
        );
        expect(wrapper).toMatchSnapshot();
    });
});
