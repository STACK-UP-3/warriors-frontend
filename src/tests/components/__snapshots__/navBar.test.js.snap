// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`*************** Testing the Navbar  component *************** Should render the Navbar page correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(Symbol.observable): [Function],
    }
  }
>
  <NavBar>
    <header>
      <nav
        className="navbar"
      >
        <div
          className="nav-wrapper"
        >
          <a
            className="sidenav-trigger show-on-med-and-down left"
            data-target="slide-out"
            href="#"
          >
            <i
              className="material-icons"
            >
              menu
            </i>
          </a>
          <a
            className="brand-logo"
            href="#"
          >
            <img
              className="responsive-img"
              src={
                Object {
                  "process": [Function],
                }
              }
            />
          </a>
          <ul
            className="right hide-on-med-and-down"
            id="nav-mobile"
          >
            <li>
              <a
                href="/dashboard"
              >
                <i
                  className="material-icons"
                >
                  home
                </i>
              </a>
            </li>
            <li>
              <a
                href="/notifications"
              >
                <i
                  className="material-icons"
                >
                  notifications
                </i>
              </a>
            </li>
            <li>
              <a
                href="/account"
              >
                <i
                  className="material-icons"
                >
                  account_circle
                </i>
              </a>
            </li>
          </ul>
        </div>
      </nav>
    </header>
  </NavBar>
</Provider>
`;
